source/Account.cpp

```diff
@@ -329,116 +321,298 @@ void Account::SetSalaryIncome(const string &name, int64_t amount)
+		int64_t payment = mortgage.Payment();
+		if(payment > credits)
+		{
+			mortgage.MissPayment();
+			receipt.paidInFull = false;
+		}
+		else
+		{
+			payment = mortgage.MakePayment();
+			credits -= payment;
+			receipt.creditsPaid += payment;
+		}
+	}
+
+	return receipt;
 }
 
 
 
-// Check how big a mortgage the player can afford to pay at their current income.
-int64_t Account::Prequalify() const
-{
-	double payments = 0.;
-	int64_t liabilities = 0;
-	for(const Mortgage &mortgage : mortgages)
+const string Account::GenerateMissedPaymentLogs(std::vector<Receipt> *receipts) const {
+	ostringstream log;
+
+	if(!receipts->at(0).paidInFull)
 	{
-		payments += mortgage.PrecisePayment();
-		liabilities += mortgage.Principal();
+		log << "You could not pay all your crew salaries. ";
 	}
 
-	// Put a limit on new debt that the player can take out, as a fraction of
-	// their net worth, to avoid absurd mortgages being offered when the player
-	// has just captured some very lucrative ships.
-	return max<int64_t>(0, min(
-		NetWorth() / 3 + 500000 - liabilities,
-		Mortgage::Maximum(YearlyRevenue(), creditScore, payments)));
+	if(!receipts->at(1).paidInFull)
+	{
+		log << "You could not pay all your maintenance costs. ";
+	}
+
+	if(Mortgages().size() > 0)
+	{
+		if(!receipts->at(2).paidInFull || !receipts->at(3).paidInFull)
+		{
+			log << "You missed a mortgage payment. ";
+		}
+	}
+
+	return log.str();
 }
 
 
 
-// Get the player's total net worth (counting all ships and all debts).
-int64_t Account::NetWorth() const
+void Account::UpdateMortgages()
 {
-	return history.empty() ? 0 : history.back();
+	for(auto it = mortgages.begin(); it != mortgages.end(); )
+	{
+		if(!it->Principal())
+			it = mortgages.erase(it);
+		else
+			++it;
+	}
 }
 
 
 
-// Find out the player's credit rating.
-int Account::CreditScore() const
+int64_t Account::CalculateNetWorth(int64_t assets) const
 {
-	return creditScore;
+	return assets - TotalDebt() - OverdueCrewSalaries() - OverdueMaintenance();
 }
 
 
 
-// Get the total amount owed for "Mortgage", "Fine", or both.
-int64_t Account::TotalDebt(const string &type) const
-{
-	int64_t total = 0;
-	for(const Mortgage &mortgage : mortgages)
-		if(type.empty() || mortgage.Type() == type)
-			total += mortgage.Principal();
+void Account::UpdateCreditScore(std::vector<Receipt> *receipts) {
+	// If you failed to pay any debt, your credit score drops. Otherwise, even
+	// if you have no debts, it increases. (Because, having no debts at all
+	// makes you at least as credit-worthy as someone who pays debts on time.)
+	bool missedPayment = false;
+	for(Receipt receipt : *receipts)
+	{
+		if(!receipt.paidInFull)
+		{
+			missedPayment = true;
+			break;
+		}
+	}
+	creditScore = max(200, min(800, creditScore + (missedPayment ? -5 : 1)));
+}
 
-	return total;
+
+
+bool Account::AnyPaymentsMade(std::vector<Receipt> *receipts) {
+	for(Receipt receipt : *receipts)
+	{
+		if(receipt.creditsPaid > 0)
+			return true;
+	}
```

Comment:
```suggestion
		if(receipt.creditsPaid > 0)
			return true;
```

Reply:
For loops should always be enclosed with curly braces

Reply:
style guide disagrees

Reply:
source?

Reply:
you can look at the style guide in the wiki, I am going to sleep now

Reply:
Says you *can*, not that you must. And for a for loop, even if it's legal C++ I'm gonna keep it encased in curly braces

Reply:
The text in the "Single-line loop bodies" entry indeed didn't sound very directive, but I'm very sure that the text `"do not need to"` really should be read as `"should not"`. Styleguide entries are not written with the intention of being optional.
I created https://github.com/endless-sky/endless-sky.github.io/pull/48 to make this entry in the styleguide more clear.